### PR TITLE
Update to Ceno's version of the toolchain

### PR DESCRIPTION
**Pull request by matthiasgoergens**
We are getting a lot of [flak from dependabot about security problems](https://github.com/scroll-tech/ceno-recursion/security/dependabot) in dependencies.

So let's update our dependencies to fix them.

Also rename to `config.toml` to make the newer cargo happy.

### CODE REVIEW

File: spartan_parallel/.cargo/config
Comment (matthiasgoergens) [inline spartan_parallel/.cargo/config:1]: This was actually just a rename to `spartan_parallel/.cargo/config.toml`

File: ff/Cargo.toml [hunk 1]
```diff
@@ -16,12 +16,12 @@ edition = "2021"
 [dependencies]
 bitvec = { version = "1", default-features = false, optional = true }
 byteorder = { version = "1", default-features = false, optional = true }
-ff_derive = { version = "0.12.1", path = "ff_derive", optional = true }
-rand_core = { version = "0.6", default-features = false }
-subtle = { version = "2.2.1", default-features = false, features = ["i128"] }
+ff_derive = { version = "0", path = "ff_derive", optional = true }
```
Comment (lispc) [inline ff/Cargo.toml:19]: seems not a good idea.   v0.x and v0.y don't need to be compatible in sem ver

```ff_derive = { version = "0",```

File: spartan_parallel/src/lib.rs [hunk 2]
```diff
@@ -974,13 +974,13 @@ impl SNARK {
     // PREPROCESSING
     // --
     // unwrap the assignments
-    let mut block_vars_mat = block_vars_mat.into_iter().map(|a| a.into_iter().map(|v| v.assignment).collect_vec()).collect_vec();
-    let mut exec_inputs_list = exec_inputs_list.into_iter().map(|v| v.assignment).collect_vec();
-    let mut init_phy_mems_list = init_phy_mems_list.into_iter().map(|v| v.assignment).collect_vec();
-    let mut init_vir_mems_list = init_vir_mems_list.into_iter().map(|v| v.assignment).collect_vec();
-    let mut addr_phy_mems_list = addr_phy_mems_list.into_iter().map(|v| v.assignment).collect_vec();
-    let mut addr_vir_mems_list = addr_vir_mems_list.into_iter().map(|v| v.assignment).collect_vec();
-    let mut addr_ts_bits_list = addr_ts_bits_list.into_iter().map(|v| v.assignment).collect_vec();
```
[the remark this answers is not in the document]
Comment (matthiasgoergens) [reply spartan_parallel/src/lib.rs]: Oh, I figured it out: `collect_vec` wasn't available by default anymore.  So we would have needed to import it.  But this was cleaner.

File: ff/Cargo.toml (hunk 1 above)
Comment (matthiasgoergens) [reply ff/Cargo.toml]: They actually need to be compatible **in semantic versioning**.  But you are right that **in practice** they often aren't.

Fixed.

**Pull request merged**